### PR TITLE
Renamings related to KeyValueStore

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

To better reflect the purpose of the code, we have renamed several traits

## Proposal

The following renames were done:
* rename `KeyValueStoreClient` to `KeyValueStore`.
* rename `*KvStore* `to `*Store*`  (all stores are key-value stores anyway)
* rename `*Client` to `*Store`
* Renaming of `ContextFromDb` to `ContextFromStore` which is clearer.
* rename the `client` to `store` accordingly.

## Test Plan

The CI should take care of it.

## Release Plan

Some typos in the documentation were identified in the process and this impacts the release.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-sdk/src/views/system_api.rs [hunk 1]
```diff
@@ -9,23 +9,23 @@ use async_trait::async_trait;
...
     views::ViewError,
 };
 
 /// We need to have a maximum key size that handles all possible underlying
 /// sizes. The constraint so far is DynamoDb which has a key length of 1024.
 /// That key length is decreased by 4 due to the use of a value splitting.
-/// Then the KeyValueStoreClient needs to handle some base_key and so we
+/// Then the KeyValueStore needs to handle some base_key and so we
 /// reduce to 900. Depending on the size, the error can occur in system_api
 /// or in the KeyValueStoreView.
```
Comment (afck) [inline linera-sdk/src/views/system_api.rs:21]:
```suggestion
/// or in the `KeyValueStoreView`.
```

File: linera-sdk/src/views/system_api.rs [hunk 2]
```diff
@@ -9,23 +9,23 @@ use async_trait::async_trait;
...
 const MAX_KEY_SIZE: usize = 900;
 
 /// A type to interface with the key value storage provided to applications.
 #[derive(Default, Clone)]
-pub struct KeyValueStore;
+pub struct WasmStore;
```
Comment (afck) [inline linera-sdk/src/views/system_api.rs:26]: (Or maybe `AppStateStore` or `ContractStateStore`?)

**Review by afck — APPROVED** [review]

File: linera-sdk/src/views/system_api.rs (hunk 2 above)
Comment (ma2bd) [reply linera-sdk/src/views/system_api.rs:26]: `SystemStore` or `Application(State)Store` work for me.

(`ContractState` is too narrow. `Wasm` is not the right concept I think)

File: linera-views/src/common.rs [hunk 3]
```diff
@@ -539,33 +539,33 @@ pub trait Context {
...
 
 /// Implementation of the [`Context`] trait on top of a DB client implementing
-/// [`KeyValueStoreClient`].
+/// [`KeyValueStore`].
 #[derive(Debug, Default, Clone)]
-pub struct ContextFromDb<E, DB> {
+pub struct ContextFromStore<E, S> {
     /// The DB client that is shared between views.
-    pub db: DB,
+    pub store: S,
```
Comment (ma2bd) [inline linera-views/src/common.rs:546]: nice, thanks!

**Review by ma2bd — APPROVED** [review]

File: linera-views/src/common.rs [hunk 4]
```diff
@@ -539,33 +539,33 @@ pub trait Context {
 }
 
 /// Implementation of the [`Context`] trait on top of a DB client implementing
-/// [`KeyValueStoreClient`].
+/// [`KeyValueStore`].
 #[derive(Debug, Default, Clone)]
-pub struct ContextFromDb<E, DB> {
+pub struct ContextFromStore<E, S> {
```
Comment (ma2bd) [inline linera-views/src/common.rs:544]: The trait could be called `ViewContext` and this struct just `Context`.

Comment (ma2bd) [reply linera-views/src/common.rs:544]: (next PR ok)

File: linera-views/src/value_splitting.rs [hunk 5]
```diff
@@ -35,15 +35,15 @@ pub enum DatabaseConsistencyError {
 /// stored as multiple smaller key-value pairs in the wrapped store.
 /// See the README.md for additional details.
 #[derive(Clone)]
-pub struct ValueSplittingKeyValueStoreClient<K> {
+pub struct ValueSplittingKeyValueStore<K> {
```
[the remark this answers is not in the document]
Comment (MathieuDutSik) [reply linera-views/src/value_splitting.rs]: Ok.

File: linera-views/src/common.rs (hunk 4 above)
Comment (MathieuDutSik) [reply linera-views/src/common.rs:544]: Yes, next PR though I am unsure about that change.

**Pull request merged**